### PR TITLE
Using url-parse also in cothority-ts

### PR DESCRIPTION
**Pull request by ineiti**

### CODE REVIEW

File: mobile/app/lib/cothority/network/proto.ts [hunk 1]
```diff
@@ -172,21 +173,22 @@ export class ServerIdentity extends Message<ServerIdentity> {
...
                 break;
             }
             case "https:": {
-                urlParser.protocol = "wss:";
+                proto = "wss:";
                 break;
             }
             default : {
                 throw new Error("The url field should use either 'http:' or 'https:', but we found "
                     + urlParser.protocol);
```
Comment (tharvik) [inline mobile/app/lib/cothority/network/proto.ts]:
```suggestion
                     + proto);
```

File: mobile/app/lib/cothority/network/proto.ts [hunk 2]
```diff
@@ -172,21 +173,22 @@ export class ServerIdentity extends Message<ServerIdentity> {
...
             }
         }
-        let result = urlParser.toString();
+        let result = url.replace(/^https?:/, proto);
```
Comment (tharvik) [inline mobile/app/lib/cothority/network/proto.ts]: If you modify `urlParser` directly, no need to string replace stuff.

File: mobile/app/lib/cothority/network/proto.ts (hunk 1 above)
Comment (ineiti) [reply mobile/app/lib/cothority/network/proto.ts]: why? This is the error-message, so `proto` is not set.

File: mobile/app/lib/cothority/network/proto.ts (hunk 2 above)
Comment (ineiti) [reply mobile/app/lib/cothority/network/proto.ts]: That was the original `new URL`, but the `url-parser` doesn't allow to set the `protocol`, this is why I changed it here to use the `proto` variable.

File: mobile/app/lib/cothority/network/proto.ts (hunk 1 above)
Comment (tharvik) [reply mobile/app/lib/cothority/network/proto.ts]: Oups, you're correct

File: mobile/app/lib/cothority/network/proto.ts (hunk 2 above)
Comment (tharvik) [reply mobile/app/lib/cothority/network/proto.ts]: Hum, it seems that there is a [URL.set](https://www.npmjs.com/package/url-parse#urlsetkey-value) to do that, maybe something like that?
```suggestion
        urlParser.set("protocol", proto);
```

**Comment by ineiti** [comment]
Screw `url-parse`, this is a job for a regexp...

**Pull request merged**